tests/fsharp/core/patterns/test.fsx

```diff
@@ -1311,6 +1311,41 @@ module StructUnionMarshalingBug =
     let buffer = Marshal.AllocHGlobal(64) // HACK: just assumed a much larger size
     Marshal.StructureToPtr<Msg1>(msg1, buffer, false) 
 
+module MatchBangSimple =
+    type CardSuit = | Hearts | Diamonds | Clubs | Spades
+    let fetchSuit () = async {
+        // do something in order to not allow optimizing things away
+        Async.Sleep 1
```

Comment:
Just noticed -- should be `do! Async.Sleep 1`.

On second thought I'm not sure this is needed at all -- the compiler can't doesn't really optimize `let foo () = async { return 42 }`, does it?